inductiva/utils/files.py

```diff
@@ -136,21 +166,37 @@ def download_from_url(url: str, unzip: bool = False) -> str:
     Returns:
         The path to the downloaded file.
     """
-    # Get archive name from url passed
-    local_path = url.split("/")[-1]
-    local_path = pathlib.Path(local_path)
-
     try:
-        logging.info("Downloading from URL to the local path: %s", local_path)
-        downloaded_to, _ = urllib.request.urlretrieve(url, filename=local_path)
+        #Get head to check the filename
+        filename = url.split("/")[-1]
+        response = requests.head(url, allow_redirects=True, timeout=3)
+
+        # Check if the 'Content-Disposition' header is present
+        content_disposition = response.headers.get("Content-Disposition")
+        if content_disposition and "filename=" in content_disposition:
+            filename = content_disposition.split("filename=")[1].strip('"')
+
+        logging.info("Downloading from URL to the local path: %s", filename)
```

Comment:
This is the main action that this function performs. We can prefix it with a bullet:

```suggestion
        logging.info("■ Downloading from URL to %s", filename)
```